heronpy/api/topology.py

```diff
@@ -194,24 +194,25 @@ def get_heron_options_from_env():
 
     Currently supports the following options natively:
 
-    - `cmdline.topologydefn.tmpdirectory`: the directory to which this
+    - `cmdline.topologydefn.tmpdirectory`: (required) the directory to which this
     topology's defn file is written
-    - `cmdline.topology.initial.state`: the initial state of the topology
-    - `cmdline.topology.name`: topology name on deployment
+    - `cmdline.topology.initial.state`: (default: "RUNNING") the initial state of the topology
+    - `cmdline.topology.name`: (default: class name) topology name on deployment
 
     Returns: map mapping from key to value
     """
-    heron_options_raw = os.environ.get("HERON_OPTIONS", None)
+    heron_options_raw = os.environ.get("HERON_OPTIONS")
     if heron_options_raw is None:
       raise RuntimeError("HERON_OPTIONS environment variable not found")
 
-    ret = {}
-    heron_opt_list = heron_options_raw.replace("%%%%", " ").split(',')
-    for opt_raw in heron_opt_list:
-      opt = opt_raw.split("=")
-      if len(opt) == 2:
-        ret[opt[0]] = opt[1]
-    return ret
+    options = {}
+    for option_line in heron_options_raw.replace("%%%%", " ").split(','):
```

Comment:
This feels a bit odd, I haven't seen the format elsewhere - is it used in Java? Maybe JSON would be better as it is just `os.environ["HERON_OPTIONS"] = json.dumps(config)` and `json.loads(os.envrion["HERON_OPTIONS"])` with `KeyError` and `ValueError` handling around the latter.

Reply:
It might be worth trying and if it works - we can use json dumps and load.

Reply:
I think I'd have to get stuck into at least all of the client libraries, heron-executor, and tests, which is a bit too much for me at the moment. I don't mind making an issue for it though.

Reply:
it does look strange. From the code it seems this is used to support space in path. And the %%%% can be found in cpp, java and python with code search.